### ISSUE TITLE
Add wait/no-wait support to Commands

### ISSUE THREAD

**Issue by tjprescott** [Azure/azure-cli#1528]
PR #1463 added support for the `... wait` command for select command groups and the `--no-wait` option for several long-running operations in those groups. This issue is for identifying and tracking which commands still need this functionality exposed.

- [x] vm create/update (PR #1463)
- [x] group delete (PR #1463)
- [x] group deployment create (PR #1463)
- [x] application-gateway create/update/delete (PR #1463)
- [x] vnet-gateway create/update (PR #1463)
- [ ] local-gateway create/update/delete
- [ ] express-route create/update/delete
- [ ] vmss create/update/delete/start/stop/deallocate/...
- [ ] vm delete/deallocate/start/stop/...

**Comment by derekbekoe** [comment]
This is needed for end of Sprint 12 right?
cc: @johanste, @yugangw-msft

**Comment by derekbekoe** [comment]
Removed milestone as this is an additive change.

**Comment by tjprescott** [comment]
I would think this is not really a GA-mandatory item. It would simply be an additive feature we can incrementally roll out over time. This issue is to track those items for which it would be nice to have.

**Comment by yangl900** [comment]
Please add support for "az vm access" as well, the command took minutes to complete, it's practically not possible to manage large set of VMs without --no-wait.

**Comment by jamescgibson** [comment]
I'd love to get --no-wait on `az disk delete`

**Comment by mrx88** [comment]
I would really like to get --no-wait for "az network nic create" and "am vm extension set" commands.

**Comment by weznagwama** [comment]
az group deployment delete --no-wait would be great, ATM there's no way to async delete a batch of deploys, have to serially do it one by one and it takes a long time.

**Comment by dcbrown16** [comment]
I have a customer request for --no-wait on `az snapshot create`. Thanks!

**Comment by tjprescott** [comment]
@yugangw-msft this issue is intended not to be closed. It is intended for users to comment when there are new commands they want wait/no-wait support for.

**Comment by goenning** [comment]
Can we have `--no-wait` on `az container start` and `az container stop`? It is already supported when creating containers, but not when restarting an existing one.

**Comment by tjprescott** [comment]
@geoning I don't see any instance when `az container stop` is a long running operation.  `az container start` can be.  Is there a particular image you use that makes container stop long-running?

**Comment by goenning** [comment]
Does `az container stop` wait until the container has gracefully stopped? Because you can never know what the container does when it receives a SIGTERM. In my use case, I don't need no-wait on stop, I just mentioned to keep it consistent.

**Comment by odonyde** [comment]
A `no-wait` flag on `az cdn endpoint purge` would be great, too. :-) Currently my deployment pipeline waits over 5 minutes for purging a single asset... :-(

**Comment by DarqueWarrior** [comment]
I would like --no-wait on
```
az sql server create
```

**Comment by ATGardner** [comment]
It will be very handy to have it available for
`az acr repository delete`

**Comment by razvangoga** [comment]
**az postgres server firewall-rule create** and **az postgres server firewall-rule delete**  could use the same flag. I'm provisioning a PgSQL db with firewall rules from azure devops and from the ~8 mins pipeline run time ~4 are just to set 5 firewall rules

**Comment by emoscardini** [comment]
If possible, please add "az sig update" to the list

**Comment by tehyatt** [comment]
I need wait for az acr network-rule add. It is not a long running operation, but it's long enough to fail in my use-case. I'm using terraform to a list of outbound IP addresses from an app service to the existing list of on-premise IPs. Terraform doesn't support modifying that list after creation of the acr (see https://github.com/terraform-providers/terraform-provider-azurerm/issues/4936). So I'm using a null resource with a provisioner. But it launches all the az commands without waiting, and so some of them fail with "Error: Error running command 'az acr network-rule add --name registryname --ip-address xxx.xxx.xxx.xxx': exit status 1. Output: ERROR: Another operation is in progress for the resource /subscriptions/****/resourceGroups/rg-name/providers/Microsoft.ContainerRegistry/registries/registryname. Please try after some time."

**Comment by GitMrZ** [comment]
Need a wait option on:
az network dns zone delete


I keep my zone data in the flat file format to feed internal DNS servers.  The delete can take a few seconds or several minutes for a 400 record zone.  Obviously do not want zone down long.  THX.

**Comment by naammeinkyahai** [comment]
Please add the ability to `--no-wait` for `az postgres db delete`

**Comment by adamrushuk** [comment]
Please add `--no-wait` support for `az image delete`?

**Comment by massimo1993** [comment]
I need a wait option for storage account. I'm having issues creating a container right after the storage account is created using az cli.

```
#!/usr/bin/env bash

az storage account create \
  --name "samassimoarm" \
  --resource-group "rgMassimoTf" \
  --location "South Central US" \
  --default-action 'Allow' \
  --sku Standard_LRS


while true; do
  PROVISIONED=`
    az storage account show \
      --name "samassimoarm" \
      --resource-group "rgMassimoTf" \
      --query [provisioningState] \
      --output tsv
  `

  [[ $PROVISIONED == 'Succeeded' ]] && break
done

CONNECTION_STRING=`
  az storage account show-connection-string \
    --name "samassimoarm" \
    --resource-group "rgMassimoTf" \
    --output tsv
`

az storage container create \
  --name "arm-templates" \
  --account-name "samassimoarm" \
  --connection-string "$CONNECTION_STRING"
```

**Comment by emibcn** [comment]
I would need `--wait` support for:
```
az iot dps enrollment-group delete
```

**Comment by sdavis3** [comment]
az vm run-command needs --no-wait option.

Example command: 
az vm run-command invoke -g rgName -n vmName --no-wait --command-id RunPowerShellScript --scripts @c:\temp\run_this_script.ps1

**Comment by zhoxing-ms** [comment]
@sdavis3 If you have new requirements, please create a [feature request](https://github.com/Azure/azure-cli/issues/new?assignees=&labels=&template=Feature_request.md&title=). We will arrange our development plan according to the priority.

**Comment by michelletaal-shell** [comment]
I'd also like to nominate az sql db import/export for a no-wait
But, I would highly suggest to actually add this flag as an option to all existing commands as a global parameter.

**Comment by darkoa-msft** [comment]
What about mandating `wait` support for any API that returns 202?
I do not understand how this issue can be open for 8 years.

**Comment by teresaritorto** [comment]
Closing -stale issue
